ambry-store/src/main/java/com.github.ambry.store/IndexSegment.java

```diff
@@ -662,78 +728,88 @@ private void readFromFile(File fileToRead, Journal journal) throws StoreExceptio
-                  && offsetInLogSegment != blobValue.getOriginalMessageOffset()
-                  && blobValue.getOriginalMessageOffset() >= startOffset.getOffset()) {
-                // we add an entry for the original message offset if it is within the same index segment
-                journal.addEntry(new Offset(startOffset.getName(), blobValue.getOriginalMessageOffset()), key);
-              }
-              journal.addEntry(blobValue.getOffset(), key);
-              sizeWritten.addAndGet(key.sizeInBytes() + valueSize);
-              numberOfItems.incrementAndGet();
-              if (offsetInLogSegment + blobValue.getSize() > maxEndOffset) {
-                maxEndOffset = offsetInLogSegment + blobValue.getSize();
-              }
-            } else {
-              logger.info(
-                  "IndexSegment : {} ignoring index entry outside the log end offset that was not synced logEndOffset "
-                      + "{} key {} entryOffset {} entrySize {} entryDeleteState {}", indexFile.getAbsolutePath(),
-                  logEndOffset, key, blobValue.getOffset(), blobValue.getSize(),
-                  blobValue.isFlagSet(IndexValue.Flags.Delete_Index));
-            }
-          }
-          endOffset.set(new Offset(startOffset.getName(), maxEndOffset));
-          logger.trace("IndexSegment : {} setting end offset for index {}", indexFile.getAbsolutePath(), maxEndOffset);
-          long crc = crcStream.getValue();
-          if (crc != stream.readLong()) {
-            // reset structures
-            keySize = KEY_SIZE_INVALID_VALUE;
-            valueSize = VALUE_SIZE_INVALID_VALUE;
-            endOffset.set(startOffset);
-            index.clear();
-            bloomFilter.clear();
-            throw new StoreException("IndexSegment : " + indexFile.getAbsolutePath() + " crc check does not match",
-                StoreErrorCodes.Index_Creation_Failure);
-          }
+          persistedEntrySize = keySize + valueSize;
+          indexSizeExcludingEntries = VERSION_FIELD_LENGTH + KEY_OR_ENTRY_SIZE_FIELD_LENGTH + VALUE_SIZE_FIELD_LENGTH
+              + LOG_END_OFFSET_FIELD_LENGTH + CRC_FIELD_LENGTH;
+          break;
+        case PersistentIndex.VERSION_2:
+          persistedEntrySize = stream.readInt();
+          valueSize = stream.readInt();
+          indexSizeExcludingEntries = VERSION_FIELD_LENGTH + KEY_OR_ENTRY_SIZE_FIELD_LENGTH + VALUE_SIZE_FIELD_LENGTH
+              + LOG_END_OFFSET_FIELD_LENGTH + CRC_FIELD_LENGTH;
           break;
         default:
           throw new StoreException("IndexSegment : " + indexFile.getAbsolutePath() + " invalid version in index file ",
               StoreErrorCodes.Index_Version_Error);
       }
+      long logEndOffset = stream.readLong();
+      if (version == PersistentIndex.VERSION_0) {
+        lastModifiedTimeSec.set(indexFile.lastModified() / 1000);
+      } else if (version == PersistentIndex.VERSION_1 || version == PersistentIndex.VERSION_2) {
```

Comment:
It can be an `else`. This code was just moved out of the switch-case and there this check made sense for V0 and V1.